code/entities/cooking/hotdog-roller/extensions/HotdogRoller.State.cs

```diff
@@ -0,0 +1,124 @@
+﻿namespace Cinema;
+
+public partial class HotdogRoller
+{
+    public enum State : int
+    {
+        BothOff,
+        BothOn,
+        FrontOn,
+        BackOn
+    }
+    public State MachineState { get; set; }
+    
+    /// <summary>
+    /// Sets initial state for the Hotdog Roller. Default is both rollers off
+    /// </summary>
+    private void SetInitState()
+    {
+        TransitionStateTo(State.BothOff);
+    }
+
+    /// <summary>
+    /// Updates the power state when one of the power switches changes state
+    /// </summary>
+    private void UpdatePowerState()
+    {
+        if (!LeftSwitch.IsOn() && !RightSwitch.IsOn())
+        {
+            TransitionStateTo(State.BothOff);
+
+            return;
+        }
+
+        if (LeftSwitch.IsOn() && RightSwitch.IsOn())
+        {
+            TransitionStateTo(State.BothOn);
+
+            return;
+        }
+
+        if (LeftSwitch.IsOn() && !RightSwitch.IsOn())
+        {
+            TransitionStateTo(State.BackOn);
+
+            return;
+        }
+
+        if (!LeftSwitch.IsOn() && RightSwitch.IsOn())
+        {
+            TransitionStateTo(State.FrontOn);
+
+            return;
+        }
+    }
+
+    /// <summary>
+    /// Handles state for the hotdog roller
+    /// </summary>
+    private void HandleState()
+    {
+        switch (MachineState)
+        {
+            case State.BothOff:
+                HandleBothOffState();
+                break;
+            case State.BothOn:
+                HandleBothOnState();
+                break;
+            case State.BackOn:
+                HandleBackOnState();
+                break;
+            case State.FrontOn:
+                HandleFrontOnState();
+                break;
+        }
+    }
+    /// <summary>
+    /// Handles transition to new state
+    /// </summary>
+    private void TransitionStateTo(State state)
```

Comment:
As mentioned in another comment, you should refactor all this logic out of here and into the `Roller` component.